lua/weapons/weapon_laserpointer.lua

```diff
@@ -0,0 +1,326 @@
+
+			render.SetMaterial(Material("sprites/bluelaser1"))
+			render.DrawBeam(origin, tr.HitPos, 2, 0, 12.5, lsrclr, false)
+			render.SetMaterial(Material("sprites/light_glow02_add_noz"))
+			render.DrawSprite(tr.HitPos, size, size, lsrclr)
+		cam.End3D()
+		--[[if tr.Entity:GetMaterial() == "func_reflective_glass" and maxbounce <= 1 then--for reflecting the laser
+			local vec = origin-tr.HitPos
+			local normal = tr.HitNormal
+			local reflect = -2*vec:Dot(normal)*normal+vec
+			local trc = util.TraceLine({
+				start = tr.HitPos,
+				endpos = tr.HitPos + reflect:Forward() * 10000,
+				filter = false
+			})
+
+			render.SetMaterial(Material("sprites/bluelaser1"))
+			render.DrawBeam(tr.HitPos, trc.HitPos, 2, 0, 12.5, lsrclr, false)
+			render.SetMaterial(Material("sprites/light_glow02_add_noz"))
+			render.DrawSprite(trc.HitPos, size, size, lsrclr)
+
+			maxbounce = maxbounce+1
+		end]]
+	end
+end
+
+function SWEP:Deploy()
+	self:RemoveLaser()
+	self:SetHoldType("pistol")
+    self.Weapon:SendWeaponAnim(ACT_VM_DRAW)
+end
+
+function SWEP:OnRemove()
+	self:RemoveLaser()
+end
+
+function SWEP:PrimaryAttack()--shoots the laser
+	if (!self:CanPrimaryAttack()) then return end
+
+	local trc = self.Owner:GetEyeTrace()
+	local lctn = Location.GetLocationNameByIndex(Location.Find(trc.Entity)):lower()
+	local loc = Location.GetLocationNameByIndex(Location.Find(self.Owner)):lower()
+
+	if (self.Owner:InTheater() and not (self.Owner:GetTheater()._AllowItems)) or loc=="trump lobby" or loc=="golf" then
+		self.Owner:PrintMessage(HUD_PRINTTALK, "[red] Stop right there! Lasers are not allowed in this area. ;authority;")
+	end
+
+	self:DrawLaser()
+	self.Weapon:TakePrimaryAmmo(self.LaserPower/2+0.1)
+
+	if IsValid(trc.Entity) and trc.Entity:IsPlayer() and trc.HitBox == 0 and not -Safe(trc.Entity) and (trc.Entity:InTheater() and not (trc.Entity:GetTheater()._AllowItems)) or lctn=="trump lobby" or lctn=="golf" then
+		self:BlindPlayer(trc.Entity)
+		self:SetNextPrimaryFire(CurTime() + 0.1)
+	else
+		self:SetNextPrimaryFire(CurTime() + 0.1)
+	end
+end
+
+function SWEP:SecondaryAttack()--changes laser color
+	if (!IsFirstTimePredicted()) then return end
+
+    if not self.LaserPower then self.LaserPower = 0 end
+    self.LaserPower = (self.LaserPower+1) % (#self.LaserColors)
+    self.Owner:SetNWFloat("LaserPower",self.LaserPower)
+end
+
+function SWEP:Reload()--reloads the batteries
+	if (!IsValid(self)) or timer.Exists("Reload"..self.Owner:SteamID64()) or self.Weapon:Clip1() >= self.Primary.ClipSize or self.Owner:GetAmmoCount(self.Primary.Ammo) == 0 then return end
+
+	timer.Create("Reload"..self.Owner:SteamID64(), 0.1, 0, function()
+		self.Owner:RemoveAmmo(100-self.Weapon:Clip1(), self.Primary.Ammo, false)
+		self.Weapon:SetClip1(100)
+		self.Owner:EmitSound("items/ammo_pickup.wav")
+		timer.Remove("Reload"..self.Owner:SteamID64())
+	end)
+end
+
+function SWEP:Think()
+	if self:Clip1() > 0 then
+		self:SetNWBool("CanDrawLaser",true)

```

Comment:
`self:SetNWBool("CanDrawLaser",self:Clip1() > 0)`

Reply:
okay thanks for helping me condense!